crates/db_views/src/post_view.rs

```diff
@@ -433,10 +476,93 @@ pub struct PostQuery<'a> {
   pub is_profile_view: bool,
   pub page: Option<i64>,
   pub limit: Option<i64>,
+  pub page_after: Option<PaginationCursorData>,
+  pub page_before_or_equal: Option<PaginationCursorData>,
 }
 
 impl<'a> PostQuery<'a> {
   pub async fn list(self, pool: &mut DbPool<'_>) -> Result<Vec<PostView>, Error> {
+    if self.listing_type == Some(ListingType::Subscribed)
```

Comment:
Check [Line 271](https://github.com/LemmyNet/lemmy/pull/3872/files#diff-2379d37fd6a1a90023506953ccdd3f4ccfd8c110f2e7ffdabd99569c235e6385R271). I don't see why any of this is necessary, as its already correctly joining / filtering by `community_follower` when the subscribed view is selected. It doesn't need or use an `in` query. So I'm not sure why the simple `where post_id > cursor` won't work if the results are already correctly filtered.

Reply:
i'll mark this resolved since it's the same convo as line 271?